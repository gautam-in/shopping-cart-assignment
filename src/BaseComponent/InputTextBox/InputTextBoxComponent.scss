/******************* InputTextBoxComponent *****************************/
@import "../../Styles/color-variable.scss";
@import "../../Styles/variable.scss";

.input-textbox {
  & .input-textbox_label {
    display: none;
    color: $LIGHT_GREY;
  }

  & .input-textbox_label--focus {
    display: block;
    color: $LIGHT_BLUE;
  }

  & .input-textbox_input {
    border: none;
    border-bottom: 2px solid $LIGHT_GREY;
    width: 100%;
    height: $MEDIUM_HEIGHT;
    padding: $MEDIUM_PADDING 0;
    font-size: $MEDIUM_FONT_SIZE;
  }

  & .input-textbox_input--focus {
    border-bottom: 2px solid $LIGHT_BLUE;
  }
}
