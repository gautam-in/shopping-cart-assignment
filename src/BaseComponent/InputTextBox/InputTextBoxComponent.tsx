import { useState } from "react";
import classNames from "classnames";
import "./InputTextBoxComponent.scss";

interface IInputTextBoxComponent {
  id: string;
  type: string;
  name?: string;
  labelAndPlaceholder?: string;
  styleClassName?: string;
}

export const InputTextBoxComponent: React.FC<IInputTextBoxComponent> = (
  props
) => {
  const { id, type, labelAndPlaceholder } = props;
  const [isFocused, setFocus] = useState<boolean>(false);
  const [placeholder, setPlaceholder] = useState<string | undefined>(
    labelAndPlaceholder
  );
  const [isFieldNotEmpty, setNotFieldNotEmpty] = useState<boolean>(false);

  const BASE_CLASSNAME: string = "input-textbox";
  const LABEL_BASE_CLASSNAME: string = classNames({
    "input-textbox_label": true,
    "input-textbox_label--focus": isFocused || isFieldNotEmpty,
  });
  const INPUT_BASE_CLASSNAME: string = classNames({
    "input-textbox_input": true,
    "input-textbox_input--focus": isFocused || isFieldNotEmpty,
  });

  function onInputTextBoxBlur(e: any) {
    if (
      e.target.value !== null &&
      e.target.value !== undefined &&
      e.target.value !== ""
    ) {
      setNotFieldNotEmpty(true);
    } else {
      setPlaceholder(labelAndPlaceholder);
    }
    setFocus(false);
  }

  function onInputTextBoxFocus(e: any) {
    setFocus(true);
    setPlaceholder(undefined);
  }

  return (
    <div className={BASE_CLASSNAME}>
      <label htmlFor={id} className={LABEL_BASE_CLASSNAME}>
        {labelAndPlaceholder}
      </label>
      <input
        id={id}
        type={type}
        placeholder={placeholder}
        className={INPUT_BASE_CLASSNAME}
        onFocus={(e) => onInputTextBoxFocus(e)}
        onBlur={(e) => onInputTextBoxBlur(e)}
      />
    </div>
  );
};
