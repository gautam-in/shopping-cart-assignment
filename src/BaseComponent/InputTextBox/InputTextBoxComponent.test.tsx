import { ShallowWrapper, shallow } from "enzyme";
import { InputTextBoxComponent, IInputTextBoxComponent } from "./InputTextBoxComponent";

describe("InputTextBoxComponent", () => {
    const tree: ShallowWrapper<IInputTextBoxComponent> = shallow(
        <InputTextBoxComponent id="button" type="text"/>
    );

    test("InputTextBoxComponent Snapshot render", () => {
        expect(tree).toMatchSnapshot();
    });

    test("Check InputTextBoxComponent has classname", () => {
        expect(tree.hasClass("input-textbox")).toEqual(true)
    });
});