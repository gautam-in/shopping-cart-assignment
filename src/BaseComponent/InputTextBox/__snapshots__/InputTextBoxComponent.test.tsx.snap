// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputTextBoxComponent InputTextBoxComponent Snapshot render 1`] = `ShallowWrapper {}`;
